src/main/java/org/jabref/logic/importer/fetcher/CiteSeer.java

```diff
@@ -45,32 +49,70 @@ public Optional<HelpFile> getHelpPage() {
+        // ADR-0014
+        URL urlForQuery;
+        try {
+            urlForQuery = getURLForQuery(luceneQuery);
+        } catch (URISyntaxException | MalformedURLException | FetcherException e) {
+            throw new FetcherException("Search URI crafted from complex search query is malformed", e);
+        }
+
+        String payload = getPayloadString(new DefaultQueryTransformer().transformLuceneQuery(luceneQuery).orElse(""));
+        return getBibEntries(urlForQuery, payload);
+    }
+
+    private List<BibEntry> getBibEntries(URL urlForQuery, String payload) throws FetcherException {
+        try {
+            JsonNode requestResponse = Unirest.post(API_URL)
+                                                     .header("authority", "citeseerx.ist.psu.edu")
+                                                     .header("accept", "application/json, text/plain, */*")
+                                                     .header("content-type", "application/json;charset=UTF-8")
+                                                     .header("origin", "https://citeseerx.ist.psu.edu")
+                                                     .body(payload)
+                                                     .asJson().getBody();
+
+            Optional<JSONArray> responses = Optional.of(requestResponse)
+                                                    .map(JsonNode::getObject)
+                                                    .filter(Objects::nonNull)
+                                                    .map(jsonResponse -> jsonResponse.optJSONArray("response"))
+                                                    .filter(Objects::nonNull);
+
+            InputStream stream = new ByteArrayInputStream(responses.orElse(new JSONArray()).toString().getBytes(StandardCharsets.UTF_8));
+            List<BibEntry> fetchedEntries = getParser().parseEntries(stream);
+//            fetchedEntries.forEach(this::doPostCleanup);
+            return fetchedEntries;
+        } catch (ParseException ex) {
+            throw new FetcherException("An internal parser error occurred while parsing CiteSeer entries, ", ex);
+        }
+    }
+
     @Override
     public URL getURLForQuery(QueryNode luceneQuery) throws URISyntaxException, MalformedURLException, FetcherException {
-        URIBuilder uriBuilder = new URIBuilder("https://citeseer.ist.psu.edu/search");
-        uriBuilder.addParameter("sort", "rlv"); // Sort by relevance
-        uriBuilder.addParameter("q", new DefaultQueryTransformer().transformLuceneQuery(luceneQuery).orElse("")); // Query
-        uriBuilder.addParameter("t", "doc"); // Type: documents
-        // uriBuilder.addParameter("start", "0"); // Start index (not supported at the moment)
-        return uriBuilder.build().toURL();
+        return new URIBuilder(API_URL).build().toURL();
+    }
+
+    private String getPayloadString(String queryString) {
```

Comment:
I think, this string needs to be created by `QueryTransformer`, which inhertis from AbstractQueryTransformer. Especially, start and end year. https://github.com/JabRef/jabref/blob/2ee5c9cf3fbb095edf71d6d9044400188ebe0c0b/src/main/java/org/jabref/logic/importer/fetcher/transformers/AbstractQueryTransformer.java#L171 needs to be overwritten.

The current query transformers create a query to be appended to the URL. You can also create a JSON which gets `POST`ed using a query transformer. Please create CiteSeerQueryTransformer including test cases for it!